xdsl/runner/arith.py

```diff
@@ -0,0 +1,50 @@
+from traitlets import Any
+from xdsl.dialects.arith import Addi, Constant, Cmpi, Muli, Subi
+from xdsl.dialects.builtin import IntegerAttr
+from xdsl.interpreter import Interpreter, InterpreterFunctions, impl, register_impls
+from xdsl.utils.exceptions import InterpretationError
+from xdsl.utils.hints import isa
+
+
+@register_impls
+class ArithFunctions(InterpreterFunctions):
+    @impl(Constant)
+    def run_constant(
+        self, interpreter: Interpreter, op: Constant, args: tuple[Any, ...]
+    ):
+        if isinstance(op.value, IntegerAttr):
+            return (op.value.value.data,)
+        raise InterpretationError(
+            f"arith.constant not implemented for {type(op.value)}"
+        )
+
+    @impl(Subi)
+    def run_subi(self, interpreter: Interpreter, op: Subi, args: tuple[Any, ...]):
+        if not isa(args, tuple[int | float, int | float]):
+            raise InterpretationError(f"arith.subi unexpected operand values.")
+        return (args[0] - args[1],)
+
+    @impl(Addi)
+    def run_addi(self, interpreter: Interpreter, op: Addi, args: tuple[Any, ...]):
+        if not isa(args, tuple[int | float, int | float]):
+            raise InterpretationError()
+        return (args[0] + args[1],)
+
+    @impl(Muli)
+    def run_muli(self, interpreter: Interpreter, op: Muli, args: tuple[Any, ...]):
+        if not isa(args, tuple[int | float, int | float]):
+            raise InterpretationError()
+        return (args[0] * args[1],)
+
+    @impl(Cmpi)
+    def run_cmpi(self, interpreter: Interpreter, op: Cmpi, args: tuple[Any, ...]):
+        if not isa(args, tuple[int | float, int | float]):
```

Comment:
my understanding is that ints get promoted to floats, letting us compare anything with anything, so maybe this is worth trying:
```suggestion
        if not isa(args, tuple[float, float]):
```

Reply:
In that case why add float? I'd say that it's better to have a file with a bunch of TODOs then just waiving away the weirdness.

Reply:
Oh, right!! It should definitely be int only :upside_down_face: I really didn't pay attention to those, again, they are more placeholders than anything else at this point.
Let's fix those low hanging things quickly though, sure